### PR TITLE
Bug 1345639 Update TopSites for the new UICollectionView ActivityStream

### PR DESCRIPTION
**Pull request by farhanpatel**
This will be merged into the as-ipad branch

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
ActivityStreamPanel.swift | 76 | Limit vertical whitespace to a single empty line. Currently 2. 
ActivityStreamPanel.swift | 115 | Limit vertical whitespace to a single empty line. Currently 3. 
ActivityStreamPanel.swift | 236 | Limit vertical whitespace to a single empty line. Currently 2. 
ActivityStreamPanel.swift | 276 | Limit vertical whitespace to a single empty line. Currently 2. 
ActivityStreamPanel.swift | 296 | Limit vertical whitespace to a single empty line. Currently 2. 
ActivityStreamTopSitesCell.swift | 239 | Limit vertical whitespace to a single empty line. Currently 2. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by farhanpatel** [comment]
There are issues with the second row of TopSites getting cut off. But I'll Fix that in a separate PR.

File: Client/Frontend/Home/ActivityStreamPanel.swift [hunk 1]
```diff
@@ -143,7 +151,11 @@ extension ActivityStreamPanel {
             case .highlights: return UITableViewAutomaticDimension
             case .topSites:
                 if traits.horizontalSizeClass == .compact && traits.verticalSizeClass == .regular {
-                    return CGFloat(Int(width / ASPanelUX.TopSiteDoubleRowRatio)) + ASPanelUX.PageControlOffsetSize
+                    if width > 320 {
```
Comment (bkmunar) [inline Client/Frontend/Home/ActivityStreamPanel.swift]: put into the struct?

File: Client/Frontend/Home/ActivityStreamTopSitesCell.swift [hunk 2]
```diff
@@ -321,7 +320,7 @@ class HorizontalFlowLayout: UICollectionViewLayout {
             verticalInsets = minimumInsets
             horizontalInsets = minimumInsets
             itemSize.width = (contentSize.width - (CGFloat(horizontalItemsCount + 1) * horizontalInsets)) / CGFloat(horizontalItemsCount)
-            itemSize.height = itemSize.width
+            itemSize.height = itemSize.width + 20
```
Comment (bkmunar) [inline Client/Frontend/Home/ActivityStreamTopSitesCell.swift]: what is the +20 for?

Comment (farhanpatel) [inline Client/Frontend/Home/ActivityStreamTopSitesCell.swift]: For the title. it appears under the icon now. I'll move this into a struct

**Comment by farhanpatel** [comment]
I'm going to leave the linting errors in until we merge into master. Most of them are near code that is commented out while we transition.

**Pull request merged**